### PR TITLE
Document environment variables and $PORT

### PR DESCRIPTION
**Pull request by kmichel-sereema**
As described in #2247, this documents the effect of the `$PORT` environment variable on the default value of `--bind` and documents the impact and priority of the environment variables in the configuration overview.

~The dead code using the environment variable `LD_LIBRARY_PATH` is also removed.~

### CODE REVIEW

**Comment by benoitc** [comment]
@kmichel-sereema Thanks for the patch!

Can you move the commit "Remove dead code" in a separate PR since the concern is different? I will look on the rest asap.

**Comment by kmichel-sereema** [comment]
Fixed, the dead code change is in #2254 :)

File: docs/source/configure.rst [hunk 1]
```diff
@@ -4,23 +4,35 @@
 Configuration Overview
 ======================
 
-Gunicorn pulls configuration information from three distinct places.
+Gunicorn pulls configuration information from five distinct places.
```
Comment (tilgovi) [inline docs/source/configure.rst]:
```suggestion
Gunicorn reads configuration information from five places.
```

File: docs/source/configure.rst [hunk 2]
```diff
@@ -4,23 +4,35 @@
...
 
-The first place that Gunicorn will read configuration from is the framework
+Gunicorn will first read environment variables for the default value of some
```
Comment (tilgovi) [inline docs/source/configure.rst]:
```suggestion
Gunicorn first reads environment variables for some
```

File: docs/source/configure.rst [hunk 3]
```diff
@@ -4,23 +4,35 @@
...
+configuration :ref:`settings <settings>`.
+
+The second place that Gunicorn will read configuration from is the framework
```
Comment (tilgovi) [inline docs/source/configure.rst]:
```suggestion
Gunicorn will then read configuration from a framework
```

File: docs/source/configure.rst [hunk 4]
```diff
@@ -4,23 +4,35 @@
...
 specific configuration file. Currently this only affects Paster applications.
 
-The second source of configuration information is a configuration file that is
-optionally specified on the command line. Anything specified in the Gunicorn
-config file will override any framework specific settings.
+The third source of configuration information is a configuration file that is
+optionally specified using command line arguments. Anything specified in the
+Gunicorn config file will override any framework specific settings.
```
Comment (tilgovi) [inline docs/source/configure.rst]: Default is `gunicorn.conf.py`. If it's present, it is used implicitly. Maybe we can make that clear here.

File: docs/source/settings.rst [hunk 5]
```diff
@@ -1164,6 +1164,10 @@ Multiple addresses can be bound. ex.::
 will bind the `test:app` application on localhost both on ipv6
 and ipv4 interfaces.
 
+If the ``PORT`` environment variable is defined, the default
+is ``['0.0.0.0:$PORT']``. If it is not defined, the default
+is ``['127.0.0.1:8000']``.
+
```
Comment (tilgovi) [inline docs/source/settings.rst]: 👍 thank you!

File: docs/source/configure.rst (hunk 4 above)
Comment (kmichel-sereema) [reply docs/source/configure.rst]: That's a good idea, I added documentation for this.

**Pull request merged**